### PR TITLE
test: add `ConjunctionContainer` test cases

### PR DESCRIPTION
**Pull request by keita-determined**
## Ticket

[ET-201]

## Description

Add `ConjunctionContainer` test cases

## Test Plan

- CI passes
- coverage should be more than 80% (ConjunctionContainer component)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[ET-201]: https://hpe-aiatscale.atlassian.net/browse/ET-201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  it('should show Select when index is 1', async () => {
+    const { user } = setup({ conjunction: Conjunction.Or, index: 1 });
+
+    const conjunctionSelect = screen.getByRole('combobox');
+    expect(screen.queryByText('Where')).not.toBeInTheDocument();
+    expect(conjunctionSelect).toBeInTheDocument();
+
+    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
+    await user.click(conjunctionSelect);
+    expect(await screen.findByRole('listbox')).toBeInTheDocument();
```
Comment (johnkim-det) [inline webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx:47]: Would prefer not just testing presence of these elements but also that selecting a new conjunction calls `onClick` with the expected value

Comment (keita-determined) [reply webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx]: fixed it

File: webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    expect(conjunctionSelect).toBeInTheDocument();
+    expect(screen.getByText(Conjunction.Or)).toBeInTheDocument();
+    expect(screen.queryByText(Conjunction.And)).not.toBeInTheDocument();
+
+    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
+    await user.click(conjunctionSelect);
+    expect(await screen.findByRole('listbox')).toBeInTheDocument();
+    expect((await screen.findByRole('listbox')).children[0]).toHaveTextContent(Conjunction.And);
+    await user.click((await screen.findAllByText(Conjunction.And))[1]);
+    expect(onClick).toBeCalledTimes(1);
```
Comment (johnkim-det) [inline webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx:53]: `toBeCalledWith`?

Comment (keita-determined) [reply webui/react/src/components/FilterForm/components/ConjunctionContainer.test.tsx]: fixed

**Review by johnkim-det — APPROVED** [review]

**Pull request merged**